de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/v1_9/reader/ReaderNodeDialog.java

```diff
@@ -23,24 +23,26 @@
 import javax.swing.Box;
 import javax.swing.BoxLayout;
 import javax.swing.JPanel;
+import org.knime.base.node.io.table.read.ReadTableNodeModel;
 import org.knime.core.node.InvalidSettingsException;
 import org.knime.core.node.NodeDialogPane;
 import org.knime.core.node.NodeSettingsRO;
 import org.knime.core.node.NodeSettingsWO;
 import org.knime.core.node.NotConfigurableException;
+import org.knime.core.node.defaultnodesettings.SettingsModelString;
 import org.knime.core.node.port.PortObjectSpec;
 import org.knime.core.node.util.FilesHistoryPanel;
 import org.knime.core.node.util.FilesHistoryPanel.LocationValidation;
 import org.knime.core.node.workflow.FlowVariable;
 
 class ReaderNodeDialog extends NodeDialogPane {
 
-  private final ReaderNodeSettings nodeSettings;
+  final SettingsModelString nodeSettings;

```

Comment:
nodeSettings should be private and we need to rename nodeSettings to filePath as this no longer describes a set of settings but a single setting (path).